cfgov/v1/tests/models/test_related_posts.py

```diff
@@ -148,6 +149,25 @@ def test_related_posts_blog_limit(self):
         self.assertNotIn('Newsroom', related_posts)
         self.assertNotIn('Events', related_posts)
 
+    def test_related_posts_and_filtering(self):
+        """
+        Tests whether related posts are retrieved if and only if they match ALL
+        of the tags on the calling page.
+        """
+
+        self.block.value['relate_posts'] = True
+        self.block.value['relate_newsroom'] = True
+        self.block.value['relate_events'] = True
```

Comment:
@Scotchester it might be useful to create another comparable test to this one where `self.block.value['and_filtering']` is set to `False`, to demonstrate that (I think) pages with both `'tag1'` and `'tag2'` tags are retrieved. That'd make it a bit more clear for people trying to understand how the test works.